packages/realm/src/app-services/App.ts

```diff
@@ -219,7 +292,8 @@ export class App<
       },
       {
         baseFilePath: baseFilePath ? baseFilePath : fs.getDefaultDirectoryPath(),
-        metadataMode: binding.MetadataMode.NoEncryption,
+        metadataMode: metadata ? toBindingMetadataMode(metadata.mode) : binding.MetadataMode.NoEncryption,
+        customEncryptionKey: metadata ? metadata.encryptionKey : undefined,
```

Comment:
I believe this works as well. If `metadata` or `metadata.encryptionKey` is not defined, `undefined` is returned.
```suggestion
        customEncryptionKey: metadata?.encryptionKey,
```